test/pubsub-in-browser.spec.js

```diff
@@ -0,0 +1,179 @@
+
+function spawnWithId (factory, callback) {
+  waterfall([
+    (cb) => factory.spawnNode(cb),
+    (node, cb) => node.id((err, res) => {
+      if (err) {
+        return cb(err)
+      }
+      node.peerId = res
+      cb(null, node)
+    })
+  ], callback)
+}
+
+if (!isNode) {
+  describe('.pubsub-browser (pubsub not supported in the browsers currently)', () => {
+    const topic = 'pubsub-tests'
+
+    let factory
+    let ipfs1
+
+    before((done) => {
+      factory = new FactoryClient()
+
+      series([
+        (cb) => spawnWithId(factory, cb)
+      ], (err, nodes) => {
+        if (err) {
+          return done(err)
+        }
+
+        ipfs1 = nodes[0]
+        done()
+      })
+    })
+
+    after((done) => {
+      factory.dismantle(done)
+    })
+
+    describe('everything errors', () => {
+      describe('Callback API', () => {
+        describe('.publish', () => {
+          it('throws an error', (done) => {
+            ipfs1.pubsub.publish(topic, 'hello friend', (err, topics) => {
+              expect(err).to.exist
+              expect(err.message).to.equal(expectedError)
+              done()
+            })
+          })
+        })
+
+        describe('.subscribe', () => {
+          const handler = () => {}
+          it('throws an error', (done) => {
+            ipfs1.pubsub.subscribe(topic, {}, handler, (err, topics) => {
+              expect(err).to.exist
+              expect(err.message).to.equal(expectedError)
+              done()
+            })
+          })
+        })
+
+        describe('.peers', () => {
+          it('throws an error', (done) => {
+            ipfs1.pubsub.peers(topic, (err, topics) => {
+              expect(err).to.exist
+              expect(err.message).to.equal(expectedError)
+              done()
+            })
+          })
+        })
+
+        describe('.ls', () => {
+          it('throws an error', (done) => {
+            ipfs1.pubsub.ls((err, topics) => {
+              expect(err).to.exist
+              expect(err.message).to.equal(expectedError)
+              done()
+            })
+          })
+        })
+      })
+
+      describe('Promise API', () => {
+        describe('.publish', () => {
+          it('throws an error', () => {
+            return ipfs1.pubsub.publish(topic, 'hello friend')
+              .catch((e) => {
+                expect(e).to.exist
+                expect(e.message).to.equal(expectedError)
+              })
+          })
+        })
+
+        describe('.subscribe', () => {
+          const handler = () => {}
+          it('throws an error', (done) => {
+            ipfs1.pubsub.subscribe(topic, {}, handler)
+              .catch((e) => {
+                expect(e).to.exist
+                expect(e.message).to.equal(expectedError)
+                done()
+              })
+          })
+        })
+
+        describe('.peers', () => {
+          it('throws an error', (done) => {
```

Comment:
I'll make this even more explicit. Although, it's all mentioned in the spec file as to why "everything errors" :)